pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go

```diff
@@ -301,8 +301,12 @@ func (w *availability) Cleanup(ctx context.Context) error {
 		log.Info("waiting for namespace deletion to complete")
 		err := wait.PollUntilContextTimeout(ctx, 15*time.Second, 20*time.Minute, true, w.namespaceDeleted)
 		if err != nil {
-			log.WithError(err).Error("error waiting for namespace to delete")
-			return err
+			if errors.Is(err, context.DeadlineExceeded) {
+				log.Errorf("Timed out after 20 minutes waiting for deleted namespace: %s, %s", w.namespaceName, err)
+				return err
+			} else {
+				log.Errorf("Encountered error while waiting for deleted namespace: %s, %s", w.namespaceName, err)
```

Comment:
Could keep a successive error count, clear the value any time we don't get an error, bail when it gets to 10 or something just so we don't spin for 15 minutes if things are really busted..

Reply:
I realize this doesn't work after thinking about your other comment. I have a new idea coming.